src/resource_manager/gateway.py

```diff
@@ -27,8 +26,26 @@
 GATEWAY_PLURAL = "gateways"
 
 
-class CreateGatewayError(Exception):
-    """Represents an error when creating the gateway resource."""
+@dataclasses.dataclass
+class GatewayState(State):
+    """A part of charm state with information required to manage gateway resource.
+
+    It consists of 3 components:
+        - GatewayResourceDefinition
+        - CharmConfig
+        - SecretResourceDefinition
+
+    Attrs:
+        gateway_name: The gateway resource's name
+        external_hostname: The configured gateway hostname.
+        gateway_class: The configured gateway class.
+        secret_resource_name_prefix: Prefix of the secret resource name.
+    """
+
+    gateway_name: str
+    external_hostname: str
+    gateway_class: str
```

Comment:
Should probably be `gateway_class_name`

Reply:
I renamed it to `gateway_class_name` and also in all other relevant places